src/main/java/org/alliancegenome/curation_api/jobs/executors/VepGeneExecutor.java

```diff
@@ -0,0 +1,169 @@
+				} catch (Exception e) {
+					e.printStackTrace();
+					history.incrementFailed();
+					addException(history, new ObjectUpdateExceptionData(dtoObject, e.getMessage(), e.getStackTrace()));
+				}
+				if (history.getErrorRate() > 0.25) {
+					Log.error("Failure Rate > 25% aborting load");
+					updateHistory(history);
+					updateExceptions(history);
+					failLoadAboveErrorRateCutoff(history);
+					return false;
+				}
+				ph.progressProcess();
+			}
+			updateHistory(history);
+			updateExceptions(history);
+			ph.finishProcess();
+		}
+		return true;
+	}
+	
+	protected void runCleanup(BulkLoadFileHistory history, String dataProviderName, List<Long> annotationIdsBefore, List<Long> annotationIdsAfter, String loadTypeString, Boolean deprecate) {
+		Log.debug("runLoad: After: " + dataProviderName + " " + annotationIdsAfter.size());
+
+		List<Long> distinctAfter = annotationIdsAfter.stream().distinct().collect(Collectors.toList());
+		Log.debug("runLoad: Distinct: " + dataProviderName + " " + distinctAfter.size());
+
+		List<Long> idsToReset = ListUtils.subtract(annotationIdsBefore, distinctAfter);
+		Log.debug("runLoad: Reset: " + dataProviderName + " " + idsToReset.size());
+
+		String countType = loadTypeString + " reset";
+		
+		long existingResets = history.getCount(countType).getTotal() == null ? 0 : history.getCount(countType).getTotal();
+		history.setCount(countType, idsToReset.size() + existingResets);
+
+		String loadDescription = dataProviderName + " " + loadTypeString + " bulk load (" + history.getBulkLoadFile().getMd5Sum() + ")";
+		
+		ProcessDisplayHelper ph = new ProcessDisplayHelper(10000);
+		ph.startProcess("Deletion/deprecation of: " + dataProviderName + " " + loadTypeString, idsToReset.size());
+		
+		for (Long id : idsToReset) {
+			try {
+				predictedVariantConsequenceService.resetGeneLevelConsequence(id, loadDescription);
+				history.incrementCompleted(countType);
+			} catch (Exception e) {
+				history.incrementFailed(countType);
+				addException(history, new ObjectUpdateExceptionData("{ \"id\": " + id + "}", e.getMessage(), e.getStackTrace()));
+			}
+			if (history.getErrorRate(countType) > 0.25) {
+				Log.error(countType + " failure rate > 25% aborting load");
+				failLoadAboveErrorRateCutoff(history);
+				break;
+			}
+			ph.progressProcess();
+		}
+		updateHistory(history);
+		updateExceptions(history);
+		ph.finishProcess();
+	}
+
+	public APIResponse runLoadApi(String dataProviderName, List<VepTxtDTO> consequenceData) {
+		List<Long> idsLoaded = new ArrayList<>();
+		BulkLoadFileHistory history = new BulkLoadFileHistory(consequenceData.size());
+		history = bulkLoadFileHistoryDAO.persist(history);
+		BackendBulkDataProvider dataProvider = null;
+		if (dataProviderName != null) {
+			dataProvider = BackendBulkDataProvider.valueOf(dataProviderName);
+		}
+		runLoad(history, dataProvider, consequenceData, idsLoaded);
+		history.finishLoad();
+		return new LoadHistoryResponce(history);
+	}
+
+}
```

Comment:
Are all these methods not able to use the generic ones?

Reply:
No, it uses custom methods for both load and cleanup